src/mtev_http.c

```diff
@@ -1804,6 +1811,7 @@ _http_construct_leader(mtev_http_session_ctx *ctx) {
         }
       }
     }
+    if (!cl_present && keys != (const char **)static_key_array) free(keys);
```

Comment:
If keys is set and it isn't the static version, it needs to be free.  If *somehow* we get there with cl_present, I don't see how that simpler condition doesn't hold true.